lib/src/object_store.rs

```diff
@@ -164,4 +174,14 @@ pub enum KeyValidationError {
     ContainsCarriageReturn,
     #[error("Keys for objects cannot contain a `\n`")]
     ContainsLineFeed,
+    #[error("Keys for objects cannot contain a `[`")]
+    ContainsLeftBracket,
+    #[error("Keys for objects cannot contain a `]`")]
+    ContainsRightBracket,
+    #[error("Keys for objects cannot contain a `*`")]
+    ContainsAsterisk,
+    #[error("Keys for objects cannot contain a `?`")]
+    ContainsQuestionMark,
+    #[error("Keys for objects cannot contain a `#`")]
+    ContainsHash,
```

Comment:
This enum is getting pretty high in cardinality. I think it is appropriate to collapse all of the "cannot contain" cases to 
```
#[error("Keys for objects cannot contain a `{1}`")
Contains(String)
```